src/components/input/input.e2e.ts

```diff
@@ -1350,25 +1359,139 @@ describe("calcite-input", () => {
     expect(await element.getProperty("value")).toBe("-123");
   });
 
-  describe("is form-associated", () => {
-    it("supports type=text", () => formAssociated("calcite-input", { testValue: "test" }));
-    it("supports type=number", () => formAssociated("<calcite-input type='number'></calcite-input>", { testValue: 5 }));
-  });
+  describe("ArrowUp/ArrowDown function as Home/End within calcite-input", () => {
+    let page;
 
-  describe("external/programmatic changes to the value", () => {
-    it("incrementing correctly updates the value after focus and blur events", async () => {
-      const page = await newE2EPage();
-      await page.setContent(html` <calcite-input type="number" value="1"></calcite-input> `);
+    beforeEach(async () => {
+      page = await newE2EPage();
+    });
+
+    it("works for type textarea", async () => {
+      await page.setContent(`<calcite-input id="text-element" type="textarea" ></calcite-input>`);
       const element = await page.find("calcite-input");
-      await element.click();
-      await page.waitForChanges;
-      await element.callMethod("blur");
-      await page.waitForChanges;
-      element.setProperty("value", "2");
+
+      await element.callMethod("setFocus");
+      await page.keyboard.type("test");
       await page.waitForChanges();
-      expect(await element.getProperty("value")).toBe("2");
-      const input = await page.find("calcite-input >>> input");
-      expect(await input.getProperty("value")).toBe("2");
+
+      await page.keyboard.press("ArrowUp");
+      await page.waitForChanges();
+
+      expect(
+        await page.$eval("calcite-input", (element: HTMLInputElement) => {
+          const textarea = element.shadowRoot.querySelector("textarea");
+          textarea.focus();
+          return textarea.selectionStart === 0;
+        })
+      ).toBeTruthy();
+
+      await page.keyboard.press("ArrowDown");
+      await page.waitForChanges();
+
+      expect(
+        await page.$eval("calcite-input", (element: HTMLInputElement) => {
```

Comment:
Since a big chunk of this function is reused a bit, can you extract it to a reusable one that returns the caret index and assert on that?